### PR TITLE
fix: add voice over support for no stores found message

### LINKED ISSUE #13735: Find a Store: "No store found" message is not vocalized
**Describe the bug**
- If a screen reader user triggers a finds no stores, the message that "no stores were found" is not vocalized by the screen reader, therefore the user doesn't have any feedback about what the result of the search is.

**Tell us the version of Spartacus**
- Library version: 4.0.0

**To Reproduce**
1. Open any Screen Reader (VO, NVDA, JAWS).
2. Go to the Find a Store page
3. Enter an incorrect postal code
4. See the message.
5. Hear the absence of vocalization.

**Expected behavior**
- Screen reader vocalizes the "No stores found in databases" message.
- Add `aria-live="assertive" aria-atomic="true"` to the `<span class="cx-no-stores">` containing the message.

**Screenshots**

Screen Shot 2021-09-15 at 1 22 17 PM

### PR DESCRIPTION
**Pull request by Matejk00**
Closes: #13735

### CODE REVIEW

**Review by saad-mo — DISMISSED** [review]

**Review by developpeurweb — CHANGES_REQUESTED** [review]
Unfortunately, this doesn't work for JAWS and NVDA. As discussed on the a11y channel, only if preloading is not possible, then let's use `role="alert"`.

**Comment by Matejk00** [comment]
I added `role="alert"` to support JAWS and NVDA.

**Comment by saad-mo** [comment]
@developpeurweb I am not sure how to test this? With VO open, I can tab to the "Use My Location" button and hit enter to trigger a search. Since geolocation is not enabled, the page eventually renders "No stores found in database..." text but VO does not detect this change, and nor is the text interactive so there is still no narration. Not sure if I need to trigger the full page narration again but that does not seem very intutive. Please advise

**Review by developpeurweb — CHANGES_REQUESTED** [review]
Sorry, I meant to convey using `role="alert"` **instead of** `aria-live="assertive"` and `aria-atomic="true"`. Pls remove those two.
**Works well with JAWS and NVDA btw**

File: feature-libs/storefinder/components/store-finder-search-result/store-finder-search-result.component.html [hunk 1]
```diff
@@ -19,9 +19,14 @@
   ></cx-store-finder-list>
   <div class="container" *ngIf="!locations?.stores.length">
     <div class="row">
-      <span class="cx-no-stores">{{
-        'storeFinder.noStoresMessage' | cxTranslate
-      }}</span>
+      <span
+        class="cx-no-stores"
+        aria-live="assertive"
```
Comment (developpeurweb) [inline feature-libs/storefinder/components/store-finder-search-result/store-finder-search-result.component.html]: Please remove aria-live="assertive" aria-atomic="true" and leave only role="alert".

File: feature-libs/storefinder/components/store-finder-stores-count/store-finder-stores-count.component.html [hunk 2]
```diff
@@ -36,9 +36,14 @@ <h2 class="cx-title">
       </div>
     </div>
     <div class="row" *ngIf="!locations?.length">
-      <span class="cx-no-stores">{{
-        'storeFinder.noStoresMessage' | cxTranslate
-      }}</span>
+      <span
+        class="cx-no-stores"
+        aria-live="assertive"
```
Comment (developpeurweb) [inline feature-libs/storefinder/components/store-finder-stores-count/store-finder-stores-count.component.html]: Please remove aria-live="assertive" aria-atomic="true" and leave only role="alert".

**Comment by developpeurweb** [comment]
> @developpeurweb I am not sure how to test this? With VO open, I can tab to the "Use My Location" button and hit enter to trigger a search. Since geolocation is not enabled, the page eventually renders "No stores found in database..." text but VO does not detect this change, and nor is the text interactive so there is still no narration. Not sure if I need to trigger the full page narration again but that does not seem very intutive. Please advise

@saad-mo type any city name in North America (or a typo) then tab to the magnifying glass and hit enter.

**Review by developpeurweb — APPROVED** [review]
Works well with JAWS and NVDA

**Review by saad-mo — APPROVED** [review]

**Pull request merged**